Qt installation link broken

I can't install kiwix-desktop because of qt.
I run `kiwix-build --target-platform native_static kiwix-desktop`.

It already has icu4c, it downloads zlib, but fails for qt:

> prepare sources icu4c :
  download icu4c : SKIP
  extract icu4c : SKIP
  patch icu4c : SKIP
prepare sources zlib :
  download zlib : OK (3.4s)
  extract zlib : OK (0.1s)
  patch zlib : OK (0.0s)
prepare sources qt :
  download qt : Sha256 for qt-everywhere-src-5.10.1.tar.xz not set, do no verify download
OK (2.2s)
  extract qt : ERROR
Traceback (most recent call last):
  File "/opt/kiwix_build/bin/kiwix-build", line 8, in <module>
    sys.exit(main())
  File "/opt/kiwix_build/lib/python3.7/site-packages/kiwixbuild/__init__.py", line 80, in main
    builder.run()
  File "/opt/kiwix_build/lib/python3.7/site-packages/kiwixbuild/builder.py", line 197, in run
    self.prepare_sources()
  File "/opt/kiwix_build/lib/python3.7/site-packages/kiwixbuild/builder.py", line 107, in prepare_sources
    source.prepare()
  File "/opt/kiwix_build/lib/python3.7/site-packages/kiwixbuild/dependencies/base.py", line 125, in prepare
    self.command('extract', self._extract)
  File "/opt/kiwix_build/lib/python3.7/site-packages/kiwixbuild/dependencies/base.py", line 76, in command
    ret = function(*args, context=context)
  File "/opt/kiwix_build/lib/python3.7/site-packages/kiwixbuild/dependencies/base.py", line 121, in _extract
    name=self.source_dir)
  File "/opt/kiwix_build/lib/python3.7/site-packages/kiwixbuild/utils.py", line 215, in extract_archive
    archive = tarfile.open(archive_path)
  File "/usr/lib64/python3.7/tarfile.py", line 1578, in open
    raise ReadError("file could not be opened successfully")
tarfile.ReadError: file could not be opened successfully

I find the archive at ./ARCHIVE/qt-everywhere-src-5.10.1.tar.xz.

    file ./ARCHIVE/qt-everywhere-src-5.10.1.tar.xz
> ./ARCHIVE/qt-everywhere-src-5.10.1.tar.xz: HTML document, UTF-8 Unicode text, with very long lines

This seems to be really an HTML document, not an xz file. I'll try to download the correct file from internet (it will take a while). But it seems that the link to qt source must be fixed somewhere in the code.

I tried to install with qt version 5.13.2. I had to rename the .xz file to qt-everywhere-src-5.10.1.tar.xz (so that the installation script worked). 
But it returned a lot of ld errors like

>/usr/bin/ld: /opt/kiwix_build/kiwix-build/BUILD_native_static/qt-5.10.1/qtdeclarative/qml/QtQuick.2/libqtquick2plugin.a(plugin.o): in function `QtQuick2Plugin::~QtQuick2Plugin()':
plugin.cpp:(.text._ZN14QtQuick2PluginD2Ev[_ZN14QtQuick2PluginD5Ev]+0x2a): undefined reference to `QQmlQtQuick2Module::undefineModule()'
/usr/bin/ld: /opt/kiwix_build/kiwix-build/BUILD_native_static/qt-5.10.1/qtdeclarative/qml/QtQuick.2/libqtquick2plugin.a(plugin.o): in function `non-virtual thunk to QtQuick2Plugin::~QtQuick2Plugin()':
plugin.cpp:(.text._ZN14QtQuick2PluginD2Ev[_ZN14QtQuick2PluginD5Ev]+0x71): undefined reference to `QQmlQtQuick2Module::undefineModule()'
/usr/bin/ld: /opt/kiwix_build/kiwix-build/BUILD_native_static/qt-5.10.1/qtdeclarative/qml/QtQuick.2/libqtquick2plugin.a(plugin.o): in function `non-virtual thunk to QtQuick2Plugin::~QtQuick2Plugin()':
plugin.cpp:(.text._ZN14QtQuick2PluginD0Ev[_ZN14QtQuick2PluginD5Ev]+0x39): undefined reference to `QQmlQtQuick2Module::undefineModule()'
/usr/bin/ld: /opt/kiwix_build/kiwix-build/BUILD_native_static/qt-5.10.1/qtdeclarative/qml/QtQuick.2/libqtquick2plugin.a(plugin.o): in function `QtQuick2Plugin::~QtQuick2Plugin()':
plugin.cpp:(.text._ZN14QtQuick2PluginD0Ev[_ZN14QtQuick2PluginD5Ev]+0x79): undefined reference to `QQmlQtQuick2Module::undefineModule()'
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:74: ../../bin/qml] Error 1
make[3]: Leaving directory '/opt/kiwix_build/kiwix-build/BUILD_native_static/qt-5.10.1/qtdeclarative/tools/qml'
make[2]: *** [Makefile:153: sub-qml-make_first] Error 2
make[2]: Leaving directory '/opt/kiwix_build/kiwix-build/BUILD_native_static/qt-5.10.1/qtdeclarative/tools'
make[1]: *** [Makefile:73: sub-tools-make_first] Error 2
make[1]: Leaving directory '/opt/kiwix_build/kiwix-build/BUILD_native_static/qt-5.10.1/qtdeclarative'
make: *** [Makefile:237: module-qtdeclarative-make_first] Error 2
Stopping build due to errors

So a simple change of the version doesn't work. I won't open a separate issue for that, because the proper version seems to be 5.10 (even though I couldn't install that). Anyway in the meantime I downloaded an old version of kiwix-xulrunner, and it works great, so my personal battle is finished for now.

--- a/kiwixbuild/dependencies/qt.py
+++ b/kiwixbuild/dependencies/qt.py
@@ -17,7 +17,7 @@ class Qt(Dependency):
         source_dir = "qt-5.10.1"
         archive = Remotefile('qt-everywhere-src-5.10.1.tar.xz',
                              '',
-                             'http://ftp1.nluug.nl/languages/qt/archive/qt/5.10/5.10.1/single/qt-everywhere-src-5.10.1.tar.xz')
+                             'http://ftp.oregonstate.edu/.1/blfs/conglomeration/qt5/qt-everywhere-src-5.10.1.tar.xz')
 
     class Builder(MakeBuilder):
         dependencies = ['icu4c', 'zlib']

